src/main/java/com/redhat/devtools/intellij/lsp4ij/LanguageServerWrapper.java

```diff
@@ -303,7 +309,11 @@ public synchronized void start() throws LanguageServerException {
                 UnaryOperator<MessageConsumer> wrapper = consumer -> (message -> {
                     logMessage(message, consumer);
                     try {
-                        consumer.consume(message);
+                        if (!ApplicationManager.getApplication().isDispatchThread()) {
+                            CompletableFuture.runAsync(() -> consumer.consume(message));
+                        } else {
+                            consumer.consume(message);
+                        }
```

Comment:
Should be 
```java
                        if (ApplicationManager.getApplication().isDispatchThread()) {
                            CompletableFuture.runAsync(() -> consumer.consume(message));
                        } else {
                            consumer.consume(message);
                        }
```

as to avoid running long-running operations on the dispatch thread

Reply:
We need to di asynch in both usecase. fixed